app/models/loan.rb

```diff
@@ -117,9 +117,15 @@ def agent_names
     [primary_name, secondary_name]
   end
 
-  def name
+  def default_name
     date = signing_date || created_at.to_date
-    "#{organization.name} &ndash; #{I18n.l date}".html_safe
+    I18n.t("loan.default_loan_name", org: organization.name, date: I18n.l(date))
+  end
+
+  def display_name
+    if self.name.blank?
+      self.default_name
+    else self.name end
```

Comment:
Actually, you don't need `self` anywhere here, and you can/should use `? :`. So `name.blank? ? default_name : name`. Please try to remember to think about when `self` is necessary. Usually only on assignment. I have pointed this out several times before.